modules/apps/commerce/commerce-frontend-js/src/main/resources/META-INF/resources/components/autocomplete/Autocomplete.js

```diff
@@ -69,26 +81,49 @@ function Autocomplete({onChange, onItemsUpdated, onValueUpdated, ...props}) {
 	}, [items, props.autofill, props.itemsKey, props.itemsLabel]);
 
 	useEffect(() => {
-		const value =
-			selectedItem && getValueFromItem(selectedItem, props.itemsKey);
+		updateSelectedItem(
+			formatAutocompleteItem(
+				props.initialValue,
+				props.itemsKey,
+				props.initialLabel,
+				props.itemsLabel
+			)
+		);
+	}, [
+		props.initialLabel,
+		props.initialValue,
+		props.itemsKey,
+		props.itemsLabel,
+	]);
```

Comment:
I would technically categorize this as [smelly code](https://en.wikipedia.org/wiki/Code_smell). It is not buggy, it does it's intended purpose, and as such is not a blocker for this PR. But, it is an indication that there are larger structural problems in the component and the way it is used.

Reply:
Yeah, in terms of behavior it will do what it intended, I really don't think it's a good solution to accomplish this behavior using `useEffect`, in which case we're dealing with it as a controlled and uncontrolled state, that's not wrong but we can have some bad effects, for example:

- When you control the above component state you have two sources of truth, you keep an internal state which is synchronized with the "initial" values only in some cases.
- Hypothetically if the values might change, you cause two unnecessary renders, one component above is controlling these values and causes a new rendering, another will happen after `useEffect` is called.
- The names also change their behavior, that is, sometimes they don't do what would be just an `initial` value.
- If the value can be updated and you also internally update the above component you will never know that this value has changed and when there is a change of the props value, you will lose the internal state changes, so in moments you may have the values not synchronized.

I don't see a problem with keeping these two behaviors, this is common but the way it's done can be different, like switching between internal and external state, the APIs should be consistent if you have a property that calls `initialValue` and it can to be updated, you have to have an `onValueChange` callback, if the `onValueCallback` is not needed because you don't change this value internally, maybe you don't even need this state internally just receiving the values would be enough.

Reply:
You have a point, guys! I would for now keep it this way to deliver the diagram and then having a better look to check these derived states and updating all the JSPs that require it.

Reply:
Sure, no problem, as Marko said, this is not a blocker for this PR, feel free to investigate it further later.